skimage/transform/radon_transform.py

```diff
@@ -124,8 +124,48 @@ def _sinogram_circle_to_square(sinogram):
     pad_width = ((pad_before, pad - pad_before), (0, 0))
     return util.pad(sinogram, pad_width, mode='constant', constant_values=0)
 
+def iradon_workspace(theta, output_size, circle=False, full=True):
+    """
+    Generate workspace needed for iradon().
+
+    This allows work arrays to be calculated once for an output size and set
+    of angles, useful for multiple calls to iradon() for the same geometry.
 
-def iradon(radon_image, theta=None, output_size=None,
+    Parameters
+    ----------
+    theta : array_like, dtype=float, optional
+        Reconstruction angles (in degrees). Default: m angles evenly spaced
+        between 0 and 180 (if the shape of `radon_image` is (N, M)).
+    output_size : int
+        Number of rows and columns in the reconstruction.
+    circle : boolean, optional
+        Assume the reconstructed image is zero outside the inscribed circle.
+        Also changes the default output_size to match the behaviour of
+        ``radon`` called with ``circle=True``.
+    full : boolean, optional
+        Whether to generate full workspace (default) or only the
+        grid data (for older behavior, smaller memory usage).
+    Returns
+    -------
+    workspace: 
+        The workspace
+    """
+    if not circle:
+        output_size = int(np.floor(np.sqrt((output_size)**2 / 2.0)))
+
+    [_x, _y] = np.mgrid[0:output_size, 0:output_size]
+    xpr = _x - int(output_size) // 2
+    ypr = _y - int(output_size) // 2
+    thw = [None]*len(theta)
+    if full:
+        th  = np.deg2rad(theta)
+        thw = np.zeros((len(th), output_size, output_size),
+                       dtype=np.float64)
+        for i in range(len(th)):
+            thw[i] = ypr * np.cos(th[i]) - xpr * np.sin(th[i])
+    return (xpr, ypr, thw)
```

Comment:
it's important that this is a public method. It is designed to be used by the end-user, prior to multiple calls of iradon().

The returned value is not meant to be inspected or altered by the end user.  It is meant to be sent to iradon() as is.   It's type is an implementation detail.

But, it's type is a tuple of ndarrays.  It is not an ndarray.  It is not a namedtuple.  So, If the type is to be specified, anything other than "tuple of ndarrays" is incorrect.  So it really has to be either "tuple of ndarrays" or unspecified.